pagespeed/kernel/base/string_util.h

```diff
@@ -17,48 +17,182 @@
  * under the License.
  */
 
-
 #ifndef PAGESPEED_KERNEL_BASE_STRING_UTIL_H_
 #define PAGESPEED_KERNEL_BASE_STRING_UTIL_H_
 
-#include <cctype>                      // for isascii
+#include <cctype> // for isascii
 #include <cstddef>
 #include <map>
 #include <set>
 #include <vector>
 
-#include "base/logging.h"
+#include <iostream>
+
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/string_view.h"
+// XXX(oschaaf): internal util
+#include "absl/strings/internal/memutil.h"
+#include "absl/strings/match.h"
+
+//#include "base/logging.h"
 #include "pagespeed/kernel/base/basictypes.h"
 #include "pagespeed/kernel/base/string.h"
 
+#include "fmt/printf.h"
 
-#include <cstdlib>  // NOLINT
+#include <cstdlib> // NOLINT
 #include <string>  // NOLINT
-#include "base/strings/string_number_conversions.h"
-#include "base/strings/string_piece.h"
-#include "base/strings/string_util.h"
-#include "base/strings/stringprintf.h"
 
-using base::StringAppendF;
-using base::StringAppendV;
-using base::SStringPrintf;
-using base::StringPiece;
-using base::StringPrintf;
+static const int32 kint32max = 0x7FFFFFFF;
+static const int32 kint32min = -kint32max - 1;
+
+using namespace absl;
```

Comment:
I would recommend avoiding this; I'm scared of what we might grow to be dependent on in the presence of this. Instead just inherit from "public absl::string_view" below.

Reply:
Fixed, this is now constrained to

```
using absl::StrCat;
using absl::StrAppend;
```

Reply:
done